### ISSUE TITLE
InvalidParameterValue: cannot cast agtype

### ISSUE THREAD

**Issue by amilenovic** [apache/age#466]
Even though I read https://age.apache.org/age-manual/master/intro/types.html , I still don't have an understanding how to work with agtype. I am obviously a novice :)

I successfully created two nodes with names: 'ONE' and 'TWO'.  I wanted to create a relationship 'REL':

```
        SELECT *
        FROM ag_catalog.cypher('test', $$
            MATCH (source), (sink)
            WHERE source.name = 'ONE' AND sink.name = 'TWO'
            CREATE (source)-[e:REL { value: 0 }]->(sink)
            RETURN e
        $$) as (e ag_catalog.agtype);
```

But I get following error:

```
sycopg.errors.InvalidParameterValue: cannot cast agtype string to type boolean
```

Can you please let me know what am I doing wrong and how to solve this problem?

Thanks!

**Comment by jrgemignani** [comment]
It works on the current master for 11.5. What environment (OS, PG, AGE versions) are you using?

```
psql-11.5-5432-pgsql=# select * from cypher('test', $$ match (u) return u $$) as (n agtype);
 n
---
(0 rows)

psql-11.5-5432-pgsql=# select * from cypher('test', $$ create (u {name:"ONE"}) return u $$) as (n agtype);
                                      n
-----------------------------------------------------------------------------
 {"id": 281474976710657, "label": "", "properties": {"name": "ONE"}}::vertex
(1 row)

psql-11.5-5432-pgsql=# select * from cypher('test', $$ create (u {name:"TWO"}) return u $$) as (n agtype);
                                      n
-----------------------------------------------------------------------------
 {"id": 281474976710658, "label": "", "properties": {"name": "TWO"}}::vertex
(1 row)

psql-11.5-5432-pgsql=# SELECT *
        FROM ag_catalog.cypher('test', $$
            MATCH (source), (sink)
            WHERE source.name = 'ONE' AND sink.name = 'TWO'
            CREATE (source)-[e:REL { value: 0 }]->(sink)
            RETURN e
        $$) as (e ag_catalog.agtype);
                                                                 e
-----------------------------------------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "REL", "end_id": 281474976710658, "start_id": 281474976710657, "properties": {"value": 0}}::edge
(1 row)

psql-11.5-5432-pgsql=#
```

**Comment by amilenovic** [comment]
Thanks for your reply @jrgemignani .

I pulled the latest AGE from docker hub: https://hub.docker.com/r/apache/age#! and use docker run, as instructed.

```
psql (11.18 (Debian 11.18-1.pgdg100+1), server 11.13 (Debian 11.13-1.pgdg100+1))

postgresDB=# \dx
                 List of installed extensions
  Name   | Version |   Schema   |         Description          
---------+---------+------------+------------------------------
 age     | 1.1.0   | ag_catalog | AGE database extension
 plpgsql | 1.0     | pg_catalog | PL/pgSQL procedural language
(2 rows)
```

**Comment by jrgemignani** [comment]
@amilenovic I can't really speak to the python driver. You could try to see if it works on the command line for that docker release to see if it is the Python driver. Or, you could try to build the latest off of the master branch, as the latest works for me. 

There will be a newer version of AGE for PG11 in about 2 weeks that includes a lot of fixes and updates to the Python driver.

**Comment by amilenovic** [comment]
@jrgemignani I think I solved the problem, even though I don't fully understand the reason.

I had to execute ```SET search_path = ag_catalog, "$user", public;```. By default, that one is not executed in AGE docker container, after loading the extension. 

That was the reason I used ```ag_catalog``` prefix. 

Now, from python client, I have to make above as a first statement, after connecting to DB. Not ideal, but solves the problem.
